model/system_metrics.go

```diff
@@ -79,6 +85,31 @@ var (
 	systemMetricsInfoSchemaKey    = bsonutil.MustHaveTag(SystemMetricsInfo{}, "Schema")
 )
 
+// Find searches the database for the system metrics object. The environment should
+// not be nil. Either the ID or full Info of the system metrics object needs to be
+// specified.
+func (result *SystemMetrics) Find(ctx context.Context) error {
+	if result.env == nil {
+		return errors.New("cannot find with a nil environment")
+	}
+
+	if result.ID == "" {
+		result.ID = result.Info.ID()
+	}
+
+	result.populated = false
+	err := result.env.GetDB().Collection(systemMetricsCollection).FindOne(ctx, bson.M{"_id": result.ID}).Decode(result)
+	if db.ResultsNotFound(err) {
+		return errors.New("could not find system metrics record in the database")
```

Comment:
use Errorf and add id

Reply:
Is the ID actually useful in this case? It's just a hash of a bunch of fields.

Reply:
Yeah for debugging it can be